d2l-table-wrapper.html

```diff
@@ -143,14 +145,110 @@
 			}
 		</style>
 		<d2l-scroll-wrapper show-actions>
-			<slot></slot>
+			<slot id="slot"></slot>
 		</d2l-scroll-wrapper>
 	</template>
 	<script>
 		Polymer({
 			is: 'd2l-table-wrapper',
+			behaviors: [
+				D2L.PolymerBehaviors.Table.LocalObserverBehavior
+			],
+			listeners: {
+				'd2l-table-local-observer': '_handleLocalObserver'
+			},
+			__applyInQueue: false,
+			__tableObserver: null,
+			attached: function() {
+				Polymer.RenderStatus.afterNextRender(this, function() {
+					this._registerTableObserver();
+					this._applyClassNames();
+				}.bind(this));
+			},
+			detached: function() {
+				if (this.__tableObserver !== null) {
+					this.__tableObserver.disconnect();
+					this.__tableObserver = null;
+				}
+			},
 			notifyResize: function() {
 				this.$$('d2l-scroll-wrapper').notifyResize();
+			},
+			_applyClassNames: function() {
+
+				// don't allow a bunch of calls to queue up
+				if (this.__applyInQueue) return;
+				this.__applyInQueue = true;
+
+				fastdom.mutate(function() {
+
+					this.__applyInQueue = false;
+
+					var table = this.queryEffectiveChildren('.d2l-table');
+					if (!table) return;
+
+					var firstRow = null;
+					var lastRow = null;
+
+					for (var i = 0; i < table.rows.length; i++) {
+						var row = table.rows[i];
+						row.classList.remove('d2l-table-row-first');
+						row.classList.remove('d2l-table-row-last');
+						if (row.offsetParent !== null) {
+							if (firstRow === null) {
+								firstRow = row;
+							}
+							lastRow = row;
+						}
+						for (var j = 0; j < row.cells.length; j++) {
+							// would be great to use classList.toggle() here,
+							// but IE11 doesn't support it
+							if (j === 0) {
+								row.cells[j].classList.add('d2l-table-cell-first');
```

Comment:
I guess we're assuming these cells are visible?  Probably ok to assume?

Reply:
Yeah at this point we didn't care so I don't care. This is already a bit smarter than it was in that it doesn't require consumers to put the `last-row` attribute on the last non-invisible row. If we need to support cells, we can add that later.